lib/rubocop/cop/rspec/overwriting_setup.rb

```diff
@@ -44,18 +42,25 @@ def on_block(node)
 
         def find_duplicates(node)
           setup_expressions = Set.new
-          node.each_child_node do |child|
-            next unless setup?(child)
+          node.each_child_node(:block) do |child|
+            next unless common_setup?(child)
 
             name = if child.send_node.arguments?
-                     child.send_node.first_argument.value
+                     child.send_node.first_argument.to_a.first.to_sym
                    else
                      :subject
                    end
 
             yield child, name unless setup_expressions.add?(name)
           end
         end
+
+        def common_setup?(node)
+          return false unless setup?(node)
+          # Search only for setup with basic_literal arguments (e.g. :sym, :str)
+          # or no arguments at all.
+          node.send_node.arguments.all?(&:basic_literal?)
```

Comment:
Because of what you mentioned here https://github.com/rubocop-hq/rubocop-rspec/issues/624#issuecomment-395403485 will this also skip strings?

Reply:
It will check str but not dstr or xstr